corehq/form_processor/interfaces.py

```diff
@@ -0,0 +1,102 @@
+from couchdbkit import ResourceNotFound
+from dimagi.utils.couch.undo import DELETED_SUFFIX
+from casexml.apps.case.models import CommCareCase
+
+from couchforms.util import process_xform
+from couchforms.models import doc_types, XFormInstance, XFormError
+from couchforms.exceptions import UnexpectedDeletedXForm
+
+from .utils import to_generic
+
+
+class FormProcessorInterface(object):
+    """
+    The FormProcessorInterface serves as the base transactions that take place in forms. Different
+    backends can implement this class in order to make common interface.
+    """
+
+    @staticmethod
+    @to_generic
+    def create_from_generic(generic_xform, generic_attachment=None):
+        xform = XFormInstance.from_generic(generic_xform)
+        xform.save()
+        if generic_attachment:
+            xform.put_attachment(**generic_attachment.to_json())
+        return xform
+
+    @staticmethod
+    def get_attachment(xform_id, attachment_name):
+        return XFormInstance.get_db().fetch_attachment(xform_id, attachment_name)
+
+    @staticmethod
+    def archive_xform(xform_generic, user=None):
+        xform = FormProcessorInterface._get_xform(xform_generic.id)
+        return xform.archive(user=user)
+
+    @staticmethod
+    def unarchive_xform(xform_generic, user=None):
+        xform = FormProcessorInterface._get_xform(xform_generic.id)
+        return xform.unarchive(user=user)
+
+    @staticmethod
+    def get_xml_element(xform_generic):
+        xform = FormProcessorInterface._get_xform(xform_generic.id)
+        return xform.get_xml_element()
+
+    @staticmethod
+    @to_generic
+    def get_xform(xform_id):
+        return FormProcessorInterface._get_xform(xform_id)
```

Comment:
This is a nitpick, but I think it makes more sense to have this be a `@classmethod` and reference `FormProcessorInterface` as `cls`. (There's the conventions argument, and then there's the "what happens when you subclass this" argument, that may actually end up being relevant in this case, or maybe not, depending on how exactly we implement the two backends.)